Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -250,14 +255,37 @@ void ViewTransition::captureOldState()
         // FIXME: Fill in the rest of CapturedElement.
         CapturedElement capture;
         auto transitionName = element->computedStyle()->viewTransitionName();
-        m_namedElements.set(transitionName->name, capture);
+        m_namedElements.add(transitionName->name, capture);
     }
+    return { };
 }
 
 // https://drafts.csswg.org/css-view-transitions/#capture-new-state-algorithm
-void ViewTransition::captureNewState()
+ExceptionOr<void> ViewTransition::captureNewState()
 {
-    // FIXME: Implement this.
+    if (!m_document)
+        return { };
+    ListHashSet<AtomString> usedTransitionNames;
+    Ref document = *m_document;
+    CheckVisibilityOptions visibilityOptions { .contentVisibilityAuto = true };
+    // FIXME: Loop should probably use flat tree.
```

Comment:
The spec asks for paint order, which means we need to resolve `z-index` property too.

It's also not entirely clear that this request is well defined, since a given element might paint its background behind other elements, and its foreground in front.

Reply:
In the "capture new state" algorithm, the paint order doesn't matter too much, since we've populated most of the map.

In the "capture old state" algorithm, the plan of record was to iterate through all elements in DOM order, and collect them into a list. Then sort the elements by RenderLayer order.